From d10a87b81fa5f39a264427dcdad6f229213d9310 Mon Sep 17 00:00:00 2001
From: Samsuik <40902469+Samsuik@users.noreply.github.com>
Date: Sat, 8 Jul 2023 19:08:54 +0100
Subject: [PATCH] Fix dropping ticks when idle

If your catchup time is lower than 2 milliseconds it will
wait that time every tick without removing it from the catchup
time, this can cause tps to drop slightly when the server is idle.

diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 3e28eab9d..2eb2a14bf 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -585,6 +585,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
                         // TacoSpigot start - fix the tick loop improvements
                         if (catchupTime < 2E6) {
                             wait += Math.abs(catchupTime);
+                            catchupTime = 0; // Blossom
                         } else if (wait < catchupTime) {
                             catchupTime -= wait;
                             wait = 0;
-- 
2.40.0.windows.1

